### PR TITLE
feat: new address fix readdress topic

### PR DESCRIPTION
**Pull request by jvandaal**

### CODE REVIEW

File: src/AddressRegistry.Producer/Infrastructure/Modules/ApiModule.cs [hunk 1]
```diff
@@ -77,26 +77,6 @@ private void RegisterProjections(ContainerBuilder builder)
                 .RegisterProjectionMigrator<ProducerContextMigrationFactory>(
                     _configuration,
                     _loggerFactory)
-                .RegisterProjections<ProducerProjections, ProducerContext>(() =>
```
Comment (jvandaal) [inline src/AddressRegistry.Producer/Infrastructure/Modules/ApiModule.cs:80]: To-do: register new producer

File: src/AddressRegistry.Producer/ProducerMigrateReaddressFixProjections.cs [hunk 2]
```diff
@@ -2,190 +2,33 @@ namespace AddressRegistry.Producer
...
-    using AddressDomain = Address.Events;
+    using SqlStreamStore;
+    using SqlStreamStore.Streams;
+    using StreetName;
     using StreetName = StreetName.Events;
 
-    [ConnectedProjectionName("Kafka producer")]
-    [ConnectedProjectionDescription("Projectie die berichten naar de kafka broker stuurt.")]
-    public class ProducerProjections : ConnectedProjection<ProducerContext>
+    [ConnectedProjectionName("Kafka producer start vanaf migratie")]
```
Comment (jvandaal) [inline src/AddressRegistry.Producer/ProducerMigrateReaddressFixProjections.cs:18]: To-do: change Name and Description?

**Review by rikdepeuter — APPROVED** [review]

File: src/AddressRegistry.Producer/ProducerMigrateReaddressFixProjections.cs [hunk 3]
```diff
@@ -351,36 +185,86 @@ public ProducerProjections(IProducer producer)
...
             });
+
+            When<Be.Vlaanderen.Basisregisters.ProjectionHandling.SqlStreamStore.Envelope<StreetName.AddressHouseNumberWasReaddressed>>(async (_, message, ct) =>
+            {
+                // Always produce the old AddressHouseNumberWasReaddressed event for a single house number address.
+                await Produce(message.Message.StreetNamePersistentLocalId, message.Message.ToContract(), message.Position, ct);
+
+                // Look-up if we need to produce the missing, new StreetNameWasReaddressed event.
+                var houseNumberWasReaddressedPositions = AddressHouseNumberWasReaddressedPositionsToStreetNameWasReaddressed.List
+                    .SingleOrDefault(x => x.EndPosition != message.Position);
```
Comment (ArneD) [inline src/AddressRegistry.Producer/ProducerMigrateReaddressFixProjections.cs:196]: Shouldn't this be `==` ?

**Pull request merged**